src/cli/commands/beacon.ts

```diff
@@ -1,55 +1,90 @@
 import {CliCommand} from "./interface";
-import * as commander from "commander";
+import {CommanderStatic} from "commander";
+import {parse, JsonMap} from "@iarna/toml";
+import fs from "fs";
 import logger from "../../logger";
-import BeaconNode from "../../node";
-import defaults from "../../node/defaults";
+import BeaconNode, {BeaconNodeCtx} from "../../node";
 import {ethers} from "ethers";
 import {CliError} from "../error";
-import {IApi, IApiConstructor} from "../../rpc/api/interface";
-import {WSServer} from "../../rpc/transport";
+import {IApiConstructor} from "../../rpc/api/interface";
 import * as RPCApis from "../../rpc/api";
-import {JSONRPC} from "../../rpc/protocol";
+import defaults from "../../node/defaults";
+import deepmerge from "deepmerge";
+
+interface IBeaconCommandOptions {
+  db: string;
+  depositContract: string;
+  eth1RpcUrl: string;
+  rpc: string;
+  configFile: string;
+}
+
+interface IConfigFile extends JsonMap{
+  db?: {name: string};
+  chain?: {chain: string};
+  rpc?: {port: string};
+  eth1?: {contract?: {address: string}};
+}
 
 export class BeaconNodeCommand implements CliCommand {
 
-  public register(commander: commander.CommanderStatic): void {
+  public register(commander: CommanderStatic): void {
     commander
       .command("beacon")
       .description("Start lodestar node")
-      .option("-d, --db [db_path]", "Path to file database", defaults.db.name)
-      .option("-c, --depositContract [address]", "Address of deposit contract", defaults.eth1.depositContract.address)
+      .option("-d, --db [db_path]", "Path to file database")
+      .option("-dc, --depositContract [address]", "Address of deposit contract")
       .option("-eth1, --eth1RpcUrl [url]", "Url to eth1 rpc node")
       .option("--rpc [api]", "Exposes the selected RPC api, must be comma separated")
+      .option("-c, --configFile [config_file]", "Config file path")
       .action(async (options) => {
-        //library is not awaiting this method so don't allow error propagation (unhandled promise rejections
+        // library is not awaiting this method so don't allow error propagation
+        // (unhandled promise rejections)
         try {
           await this.action(options);
         } catch (e) {
           logger.error(e.message);
         }
-
       });
   }
 
-  public async action(options: any): Promise<void> {
-    const node = new BeaconNode({
+  public async action(options: IBeaconCommandOptions): Promise<void> {
+    let parsedConfig: IConfigFile;
+    if (options.configFile) {
+      let data: Buffer;
```

Comment:
I think it's going to be easier to test if you move config file loading and parsing into separate util class

Reply:
Great suggestion, I was planning on doing this especially since the 2 code smells relating to this function. The other 3 code smells seems very wrong to fix to me, is there a `mark as won't fix` equivalent on codeclimate or do you guys think I should change that in some way.

for reference the code smell was referencing these lines:

```Typescript
commander
      .command("beacon")
      .description("Start lodestar node")
      .option("-d, --db [db_path]", "Path to file database")
      .option("-dc, --depositContract [address]", "Address of deposit contract")
```

Reply:
I marked those code smells as wontfix